app/controllers/products_controller.rb

```diff
@@ -0,0 +1,110 @@
+class ProductsController < ApplicationController
+  before_action :find_product, only: [:show, :edit, :update, :retire, :review]
+  skip_before_action :require_login,  except: [:new, :edit, :retire]
+
+  def homepage
+    @products = Product.where(retired: false)
+  end
+
+  def browse_products
+    if params[:search_term] == 'category'
+      @products = Product.includes(:categories).where(categories: { id: params[:id] }, retired: false)
```

Comment:
It would be good to have methods within the model to retrieve these lists of products